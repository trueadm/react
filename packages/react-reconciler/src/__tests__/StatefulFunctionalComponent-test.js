/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

'use strict';

let React;
let ReactTestRenderer;

describe.only('Stateful Functional Components', () => {
  beforeEach(() => {
    jest.resetModules();
    React = require('react');
    ReactTestRenderer = require('react-test-renderer');
  });

  const StatefulFunctionalComponent = {
    initialState() {
      return {
        title: 'State!',
      };
    },
    render(props, state) {
      return <div>{props.title}. {state.title}</div>;
    },
  };

  it('renders hello world div', () => {
    const input = <StatefulFunctionalComponent title="Hello world" />;
    const tree = ReactTestRenderer.create(input).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
