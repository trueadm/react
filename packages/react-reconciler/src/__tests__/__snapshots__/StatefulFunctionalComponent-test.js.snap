// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stateful Functional Components renders hello world div 1`] = `
<div>
  Hello world
  . 
  State!
</div>
`;
